app/components/anchor/copy_to_clipboard_controller.ts

```diff
@@ -0,0 +1,29 @@
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller<HTMLDivElement> {
+  static targets = [ "initialIcon", "successIcon"];
+  static classes = [ "hidden" ]
+
+  declare readonly initialIconTarget: SVGElement & HTMLElement;
+  declare readonly successIconTarget: SVGElement & HTMLElement;
+  declare readonly notificationDelayValue: number;
+  declare readonly hiddenClass: string;
+
+  static values = {
+    notificationDelay: { type: Number, default: 1500 },
+  };
+
+  copy(): void {
+    navigator.clipboard.writeText(this.element.dataset.copyToClipboardValue || "");
```

Comment:
Not a biggie but probably this is a good use case for using [Stimulus values API](https://stimulus.hotwired.dev/reference/values)

Reply:
+1

Reply:
It also supports default values =)

Reply:
```diff
  static values = {
    notificationDelay: { type: Number, default: 1500 },
+   value: String,
  };
```

```suggestion
    if (this.hasValueValue) {
      navigator.clipboard.writeText(this.valueValue);
    }
```

Or we could rename it to read a bit better:

```diff
  static values = {
    notificationDelay: { type: Number, default: 1500 },
+   textToCopy: String,
  };
```

```suggestion
    if (this.hasTextToCopyValue) {
      navigator.clipboard.writeText(this.textToCopyValue);
    }
```